cxx/pclean/pclean.cc

```diff
@@ -31,6 +31,13 @@ T_observations translate_observations(
 
     for (size_t i = 0; i < col.second.size(); ++i) {
       const std::string& val = col.second[i];
+      if (val.empty()) {
+        // Don't incorporate missing values.
```

Comment:
Please say more, both about what needs to be done in this pull request and what needs to be done in general to handle missing values.

Reply:
Now that I think about it more, handling missing data might be simpler for the PClean-like models than for HIRM. HIRM lets you specify observed values for any relation, including a relation that serves as the base of a noisy relation. To support this in the most general way, we'd have to allow users to specify some observations of these base relations and omit other observations, so that we'd call `transition_latent_value` only for the unobserved values. With PClean, OTOH, we're allowing the user to input observations only for leaves of the relation-tree, that is observations of relations that never serve as the base of a different noisy relation. So we don't have to worry about the case where we have to infer latent values for some observations of a given relation but not for others (I think -- I'm not 100% confident about this). We'll eventually want to be able to sample from the posterior predictive for unobserved values, but that's straightforward to do. So I don't think this PR needs to do anything extra to handle missing values.